doc: align installing manually guide sections (#4177)
nrfconnect/sdk-nrf: doc/nrf/gs_installing.rst

To manually install the |NCS|, you must install all required tools and clone the |NCS| repositories.
See the following sections for detailed instructions.

The steps `Installing the required tools`_ and `Installing the toolchain`_ are similar to the installation steps in Zephyr's :ref:`zephyr:getting_started`.
If you already have your system set up to work with the Zephyr OS, you can skip these steps.

See :ref:`gs_installing_os` for information on the supported operating systems and Zephyr features.

.. _gs_installing_tools:

> .. rst-class:: numbered-step

Author: Roger that, thanks.

            echo 'export PATH=${HOME}/gn:"$PATH"' >> ${HOME}/.bashrc
            source ${HOME}/.bashrc

..

.. _gs_installing_west:

.. rst-class:: numbered-step

Install west
************

> To manage the combination of repositories and versions, the |NCS| uses :ref:`zephyr:west`.

Author: Rephrased.

> To manage the combination of repositories and versions, the |NCS| uses :ref:`zephyr:west`.

Reviewer: Still West instead of west.

            echo 'export PATH=${HOME}/gn:"$PATH"' >> ${HOME}/.bashrc
            source ${HOME}/.bashrc

..

> Installing west

Author: Agree.

===============

To manage the combination of repositories and versions, the |NCS| uses :ref:`zephyr:west`.

> See the :ref:`west documentation <zephyr:west>` for detailed information about the tool itself.

Author: Removed this sentence.

You only need to do this once.

.. _cloning_the_repositories_win:
.. _cloning_the_repositories:

.. rst-class:: numbered-step

Get the |NCS| code
******************

> The |NCS| consists of a set of Git repositories.

Author: Yup: `a set of :ref:`Git repositories <ncs_introduction>``

**********************

The |NCS| consists of a set of Git repositories.

Every |NCS| release consists of a combination of these repositories at different revisions.
The revision of each of those repositories is determined by the current revision of the main (or manifest) repository, `sdk-nrf`_.

.. note::
   The latest state of development is on the master branch of the `sdk-nrf`_ repository.
   To ensure a usable state, the `sdk-nrf`_ repository defines the compatible states of the other repositories.
   However, this state is not necessarily tested.
   For a higher degree of quality assurance, check out a tagged release.

   Therefore, unless you are familiar with the development process, you should always work with a specific release of the |NCS|.

The main repository, `sdk-nrf`_, contains a `west manifest file`_, :file:`west.yml`, that determines the revision of all other repositories.
This means that sdk-nrf acts as the :ref:`manifest repository <zephyr:west-multi-repo>`, while the other repositories are project repositories.

You can find additional information about the repository and development model in the :ref:`development model section <dev-model>`.

> Cloning the repositories

Reviewer 2: Same issue as above - as numbered section, we shouldn't have subsections. And updating is not a step of installing ... 😕

Author: I'm going to move all sections about updating to a new section at the end of the page.

               set GNUARMEMB_TOOLCHAIN_PATH=\ c:\\gnuarmemb

       .. group-tab:: Linux

          Open a terminal window and enter the following commands (assuming that you have installed the toolchain to ``~/gnuarmemb``; if not, change the value for GNUARMEMB_TOOLCHAIN_PATH):

            .. parsed-literal::
              :class: highlight

              export ZEPHYR_TOOLCHAIN_VARIANT=gnuarmemb
              export GNUARMEMB_TOOLCHAIN_PATH=\ "~/gnuarmemb"

       .. group-tab:: macOS

          Open a terminal window and enter the following commands (assuming that you have installed the toolchain to ``~/gnuarmemb``; if not, change the value for GNUARMEMB_TOOLCHAIN_PATH):

            .. parsed-literal::
              :class: highlight

              export ZEPHYR_TOOLCHAIN_VARIANT=gnuarmemb
              export GNUARMEMB_TOOLCHAIN_PATH=\ "~/gnuarmemb"

#. Set the environment variables persistently.
   Depending on your operating system:

    .. tabs::

       .. group-tab:: Windows

          Add the environment variables as system environment variables or define them in the ``%userprofile%\zephyrrc.cmd`` file as described in :ref:`build_environment_cli`.

>           This will allow you to avoid setting them every time you open a command prompt.

Author: Rephrased to `This lets you avoid`.

***********************************

Before you start :ref:`building and programming a sample application <gs_programming>`, you must set up your build environment.

1. Set up the SES environment.
   If you plan to :ref:`build with SEGGER Embedded Studio <gs_programming_ses>`, the first time you import an |NCS| project, SES might prompt you to set the paths to the Zephyr Base directory and the GNU ARM Embedded Toolchain.
   This must be done only once.
   Complete the following steps:

   a. Run the file :file:`bin/emStudio`.

   #. Select :guilabel:`File` -> :guilabel:`Open nRF Connect SDK Project`.

      .. figure:: images/ses_open.png
         :alt: Open nRF Connect SDK Project menu

         Open nRF Connect SDK Project menu

   #. Set the Zephyr Base directory to the full path to ``ncs/zephyr``.
      The GNU ARM Embedded Toolchain directory is the directory where you installed the toolchain (for example, ``c:/gnuarmemb``).

      .. figure:: images/ses_notset.png
         :alt: Zephyr Base Not Set prompt

         Zephyr Base Not Set prompt

#. Set up executables.
   The process is different depending on your operating system.

   .. tabs::

      .. group-tab:: Windows

>          Make sure the locations of tools are added to the PATH variable.

Author: Will be updated in https://projecttools.nordicsemi.no/jira/browse/NCSDK-8941

You must also make sure that nrfjprog (part of the `nRF Command Line Tools`_) is installed and its path is added to the environment variables.
The west command programs the kit by using nrfjprog by default.
For more information on nrfjprog, see `Programming SoCs with nrfjprog`_.

.. _gs_updating:

Updating tools and repositories
*******************************

Remember to regularly check for updates to tools.
Like any other Python package, the west tool is updated regularly.

You might also want to switch to a newer release or check out the latest state of development.
However, if you work with a specific release of the |NCS|, you do not need to update your repositories, because the release will not change.

.. _west_update:

Updating west
=============

To update west, run the following command:

.. tabs::

   .. group-tab:: Windows

      .. parsed-literal::
         :class: highlight

>          pip3 install -U west

Reviewer: Does this always update west to its latest version? I updated mine with pip install west == 0.10.0, for example.

Author: To its latest version available in PyPI.

   .. group-tab:: Linux

      .. parsed-literal::
         :class: highlight

         pip3 install --user -U west

   .. group-tab:: macOS

      .. parsed-literal::
         :class: highlight

         pip3 install -U west

..

Updating the repositories
=========================

To manage the ``nrf`` repository (the manifest repository), use Git.
To make sure that you have the latest changes, run ``git fetch origin`` to :ref:`fetch the latest code <dm-wf-update-ncs>` from the `sdk-nrf`_ repository.
Checking out a branch or tag in the ``nrf`` repository gives you a different version of the manifest file.

> Running ``west update`` will then update the project repositories to the state specified in this manifest file.

Reviewer: ```suggestion
Running ``west update`` updates the project repositories to the state specified in this manifest file.
```

.. include:: gs_installing.rst
   :start-after: west-error-start
   :end-before: west-error-end

For example, to switch to release |release| of the |NCS|, enter the following commands in the ``ncs/nrf`` directory:

.. parsed-literal::

>    :class: highlight

Author: You mean at the end of the line 623? Correct me if I'm wrong, but the `::` defaults to `.. code-block::` and it can also be easily mistaken as a typo by a zealous editor (not necessarily tech writer).

However, you should point them directly to their new locations as described in this section.

The full list of repositories with their old and new URLs can be found in the following table:

.. list-table::
   :header-rows: 1

   * - Old URL
     - New URL
     - File system path

   * - https:\ //github.com/NordicPlayground/fw-nrfconnect-nrf
     - https://github.com/nrfconnect/sdk-nrf
     - :file:`ncs/nrf`

   * - https:\ //github.com/NordicPlayground/fw-nrfconnect-zephyr
     - https://github.com/nrfconnect/sdk-zephyr
     - :file:`ncs/zephyr`

   * - https:\ //github.com/NordicPlayground/fw-nrfconnect-mcuboot
     - https://github.com/nrfconnect/sdk-mcuboot
     - :file:`ncs/bootloader/mcuboot`

   * - https:\ //github.com/NordicPlayground/fw-nrfconnect-mcumgr
     - https://github.com/nrfconnect/sdk-mcumgr
     - :file:`ncs/modules/lib/mcumgr`

   * - https:\ //github.com/NordicPlayground/nrfxlib
     - https://github.com/nrfconnect/sdk-nrfxlib
     - :file:`ncs/nrfxlib`

Before you change the remotes, rename any personal forks that you have of the
|NCS| repositories to their new names.

> To do so, visit your personal fork in a browser and edit the name there.

Reviewer: Numbered steps perhaps? I mean 1. edit the remote name in browser and 2. rename the actual remotes

Author: Updated.

.. _build_environment:

.. rst-class:: numbered-step

Setting up the build environment
********************************

> 

Reviewer 2: More step subsections below ... :(

Author: Need to think a second about how to handle subsections in this section, which is also not exactly the same in GSA.

Author: @ru-fu , please have another look. I've flattened out the structure of the page, so that there are no subsections in the main installation sections. I also moved the updating topics to the very end and moved the "setting up the command line environment" from subsection to an optional section after the SES one (command line sections is not present in GSA, so I think this is the way to go). But I have two doubts:
- I'm not convinced about the Updating section at the end -- Can it exist there? Would it make sense to make a separate page out of it?
- One of the links in Installing the toolchain points to the command line section, which is now optionally placed at the end, which breaks the flow -- Should we make this section mandatory and merge it with the SES one? Or rework the instruction that includes the link, so that we don't have to link at all?

Author: Discussed during NCS tech writer sync on Apr 1. Tracking task for creating an Updating page within Getting started: https://projecttools.nordicsemi.no/jira/browse/NCSDK-8936
